datalad/distribution/get.py

```diff
@@ -49,53 +51,209 @@
+    # loop over submodules not subdatasets to get the url right away
+    # install using helper that give some flexibility regarding where to
+    # get the module from
+    for sub in ds.repo.get_submodules():
+        subds = Dataset(opj(ds.path, sub.path))
+        if start is not None and not subds.path.startswith(_with_sep(start)):
+            # this one we can ignore, not underneath the start path
+            continue
+        if not subds.is_installed():
+            try:
+                subds = _install_subds_from_flexible_source(
+                    ds, sub.path, sub.url)
+                # we want the entire thing, but mark this subdataset
+                # as automatically installed
+                content_by_ds[subds.path] = [curdir]
+            except Exception as e:
+                # skip, if we didn't manage to install subdataset
+                lgr.warning(
+                    "Installation of subdatasets %s failed, skipped", subds)
+                lgr.debug("Installation attempt failed with exception: %s",
+                          exc_str(e))
+                continue
+            # otherwise recurse
+            # we can skip the start expression, we know we are within
+            content_by_ds.update(_recursive_install_subds_underneath(
+                subds,
+                recursion_limit=None if recursion_limit is None else recursion_limit - 1
+            ))
+    return content_by_ds
+
+
+def _unwind(generators):
+    res = []
+    for item in generators:
+        res.extend(item)
+    return res
+
+
 class Get(Interface):
-    """Get data content for files/directories of a dataset.
+    """Get any dataset content (files/directories/subdatasets).
 
-    Known data locations for each requested file are evaluated and data are
-    obtained from the best/fastest/cheapest location, unless a dedicated
-    source is identified.
+    This command only operates on dataset content. To obtain a new independent
+    dataset from some source use the `install` command.
 
     By default this command operates recursively within a dataset, but not
     across potential subdatasets, i.e. if a directory is provided, all files in
     the directory are obtained. Recursion into subdatasets is supported too. If
-    enabled, potential subdatasets are detected and installed sequentially, in
-    order to fulfill a request. However, this implicit installation of
-    subdatasets is done only, if an explicitly specified path belongs to such a
-    subdataset and for that purpose `recursion_limit` is ignored. Otherwise get
-    recurses into already installed subdatasets only and the depth of this
-    recursion can be limited by `recursion_limit`.
+    enabled, relevant subdatasets are detected and installed in order to
+    fulfill a request.
+
+    Known data locations for each requested file are evaluated and data are
+    obtained from the best/fastest/cheapest location, unless a specific
+    source is identified.
```

Comment:
ATM we don't have any of such (best/fastest/...) evaluation in place AFAIK, so I wouldn't promise it here